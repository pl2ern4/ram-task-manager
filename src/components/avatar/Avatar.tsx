import { styled } from '@mui/material/styles';
import Avatar from '@mui/material/Avatar';
import AccountCircleRoundedIcon from '@mui/icons-material/AccountCircleRounded';

function stringAvatar(name: string) {
    return {
      sx: {
        bgcolor: stringToColor(name),
      },
      children: `${name.split(' ')[0][0]}${name.split(' ')[1][0]}`,
    };
  }
  function stringToColor(string: string) {
    let hash = 0;
    let i;
  
    /* eslint-disable no-bitwise */
    for (i = 0; i < string.length; i += 1) {
      hash = string.charCodeAt(i) + ((hash << 5) - hash);
    }
  
    let color = '#';
  
    for (i = 0; i < 3; i += 1) {
      const value = (hash >> (i * 8)) & 0xff;
      color += `00${value.toString(16)}`.slice(-2);
    }
    /* eslint-enable no-bitwise */
  
    return color;
  }

const AvatarSection = styled(Avatar)`
  flex: 2;
  background-color:unset;
`;

type Props = {
    name?: string
}

const CustomAvatar = (props:Props) => {
    const {name} = props;
    if(!name){
        return <AvatarSection><AccountCircleRoundedIcon color="action" fontSize="large"/></AvatarSection>;
    }
    return <AvatarSection><Avatar {...stringAvatar(name)} /></AvatarSection>;
}

export default CustomAvatar;