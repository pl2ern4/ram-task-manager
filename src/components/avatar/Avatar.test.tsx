import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import Avatar from './Avatar';

describe('Avatar', () => {
it('renders without Name', () => {
    const {container} = render(<Avatar name="Tom HeadMan"/>)
    expect(container).toMatchSnapshot();
    })
  it('renders a heading', () => {
    const {container} = render(<Avatar name="Tom HeadMan"/>)
    expect(container).toMatchSnapshot();
  })
})