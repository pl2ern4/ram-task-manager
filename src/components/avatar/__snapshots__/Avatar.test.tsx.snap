// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar renders a heading 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-yzsj1s-MuiAvatar-root"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-11083xw-MuiAvatar-root"
    >
      TH
    </div>
  </div>
</div>
`;

exports[`Avatar renders without Name 1`] = `
<div>
  <div
    class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-yzsj1s-MuiAvatar-root"
  >
    <div
      class="MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-11083xw-MuiAvatar-root"
    >
      TH
    </div>
  </div>
</div>
`;
